ambry-rest/src/main/java/com.github.ambry.rest/NettyServer.java

```diff
@@ -80,15 +76,7 @@ public void start()
       // i.e. if there are a 1000 active connections there will be a 1000 NettyMessageProcessor instances.
       b.group(bossGroup, workerGroup).channel(NioServerSocketChannel.class)
           .option(ChannelOption.SO_BACKLOG, nettyConfig.nettyServerSoBacklog)
-          .handler(new LoggingHandler(LogLevel.DEBUG)).childHandler(new ChannelInitializer<SocketChannel>() {
-        @Override
-        public void initChannel(SocketChannel ch)
-            throws Exception {
-          for (Map.Entry<String, ChannelHandler> entry : channelHandlerInfoList.entrySet()) {
-            ch.pipeline().addLast(entry.getKey(), entry.getValue());
-          }
-        }
-      });
+          .handler(new LoggingHandler(LogLevel.DEBUG)).childHandler(channelInitializer);
```

Comment:
We were always reusing the same instance of `ChannelInitializer` and `LoggingHandler`. Note that `ChannelInitializer` is an object that generates new instances of all the handlers that form the pipeline when new connections are created.